autoscaler/controllers/datacollection/daemonset.go

```diff
@@ -179,7 +184,15 @@ func getDesiredDaemonSet(datacollection *odigosv1.CollectorsGroup, configData st
 	maxUnavailable := intstr.FromString("50%")
 	// maxSurge is the number of pods that can be created above the desired number of pods.
 	// we do not want more then 1 datacollection pod on the same node as they need to bind to oltp ports.
-	maxSurge := intstr.FromInt(0)
+	rollingUpdate := &appsv1.RollingUpdateDaemonSet{
+		MaxUnavailable: &maxUnavailable,
+	}
+	cfg := ctrl.GetConfigOrDie()
+	versionSupported, err := isVersionSupported(cfg, 1, 22)
```

Comment:
`v1.22.0` should be put in a variable of type version with a name suggesting what feature is supported from that version